neutrino.go

```diff
@@ -935,6 +964,33 @@ func NewChainService(cfg Config) (*ChainService, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	sideload := &sideloader{}
```

Comment:
So I think we want to be able to support _both_ side loaders active at once.

Reply:
goroutines?

Reply:
The current design is to sideload block headers if available or sideload cfilter headers if available. The two should function independently but in the case both are present, we would want to sideload blockheaders before cfheaders since cfheaders depend on block headers for verification (in case we are verifying).

Reply:
If we want the two processes to be active at once then we would have to introduce goroutines? Just like in the bloackmanager.

Reply:
By both, I just mean one after the other. The control flow in this section is very hard to follow. We can improve that with the suggestions I made re better encapsulation.